plugins/prometheus/pkg/data/data.go

```diff
@@ -0,0 +1,179 @@
+		matched, err := regexp.Match(fmt.Sprintf("^%s-([a-z0-9]{5}|[a-z0-9]{9}|[0-9]*)(-[a-z0-9]{5})?$", workload.TopController.GetName()), []byte(podName))
+		if err != nil {
+			logrus.Error(err)
+			return
+		}
+		if matched {
+			// Weak match for a pod. Don't return yet in case there's a better match.
+			name = workload.TopController.GetName()
+			kind = workload.TopController.GetKind()
+		}
+	}
+	return
+}
+
+// GetMetrics returns the memory/cpu and requests for each container running in the cluster.
+func GetMetrics(ctx context.Context, dynamicClient dynamic.Interface, restMapper meta.RESTMapper, api prometheusV1.API) ([]CombinedRequest, error) {
+	r := getRange()
+	memory, err := getMemory(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	cpu, err := getCPU(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	memoryRequest, err := getMemoryRequests(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	cpuRequest, err := getCPURequests(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	memoryLimits, err := getMemoryLimits(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	cpuLimits, err := getCPULimits(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	combinedRequests := make(map[string]CombinedRequest)
+	for _, cpuVal := range cpu {
+		key := getKey(cpuVal)
+		request := combinedRequests[key]
+		request.cpu = cpuVal.Values
+		request.Owner = getOwner(cpuVal)
+		combinedRequests[key] = request
+	}
+	for _, memVal := range memory {
+		key := getKey(memVal)
+		request := combinedRequests[key]
+		request.memory = memVal.Values
+		request.Owner = getOwner(memVal)
+		combinedRequests[key] = request
+	}
+	for _, cpuVal := range cpuRequest {
+		key := getKey(cpuVal)
+		request := combinedRequests[key]
+		request.Owner = getOwner(cpuVal)
+		request.cpuRequest = cpuVal.Values[0].Value
+		combinedRequests[key] = request
+	}
+	for _, memVal := range memoryRequest {
+		key := getKey(memVal)
+		request := combinedRequests[key]
+		request.Owner = getOwner(memVal)
+		request.memoryRequest = memVal.Values[0].Value
+		combinedRequests[key] = request
+	}
+	for _, cpuVal := range cpuLimits {
+		key := getKey(cpuVal)
+		request := combinedRequests[key]
+		request.Owner = getOwner(cpuVal)
+		request.cpuLimit = cpuVal.Values[0].Value
+		combinedRequests[key] = request
+	}
+	for _, memVal := range memoryLimits {
+		key := getKey(memVal)
+		request := combinedRequests[key]
+		request.Owner = getOwner(memVal)
+		request.memoryLimit = memVal.Values[0].Value
+		combinedRequests[key] = request
+	}
+	requestArray := make([]CombinedRequest, 0, len(combinedRequests))
+	workloads, err := controller.GetAllTopControllers(ctx, dynamicClient, restMapper, "")
+	if err != nil {
+		return nil, err
+	}
+	for _, val := range combinedRequests {
+		val.ControllerName, val.ControllerKind = getController(workloads, val.PodName, val.ControllerNamespace)
```

Comment:
It would probably be more efficient to create a map[podName]controller, rather than iterating through `workloads` every time. That array could get huge in big clusters.